financial-connections/src/main/java/com/stripe/android/financialconnections/FinancialConnectionsSheetViewModel.kt

```diff
@@ -171,15 +167,20 @@ internal class FinancialConnectionsSheetViewModel @Inject constructor(
         }
     }
 
-    private fun buildHostedAuthUrl(
-        hostedAuthUrl: String?,
-        isInstantDebits: Boolean,
-        linkMode: LinkMode?,
-    ): String? {
+    private fun buildHostedAuthUrl(hostedAuthUrl: String?): String? {
         if (hostedAuthUrl == null) {
             return null
         }
 
+        val isInstantDebits = stateFlow.value.isInstantDebits
+        val elementsSessionContext = initialState.initialArgs.elementsSessionContext
+        val linkMode = elementsSessionContext?.linkMode
+
+        val prefillDetails = elementsSessionContext?.prefillDetails
+        val email = prefillDetails?.email
```

Comment:
nit - we can likely inline this when adding query params (unless it makes them 2-liners)